regression_detector.py

```diff
@@ -188,30 +215,41 @@ def process_regressions_into_gh_issue(regressions_dict, owner: str, output_path:
         process_regressions_into_yaml(regressions_dict, output_path, args.control, args.treatment)
-    elif not args.control and args.treatment:
-        if not args.platform:
-            raise ValueError("A platform must be specified with the --platform flag to retrieve the "
-                             "previous metrics JSONs as control from S3.")
-        # Download control from S3
+        exit(0)
+
+    # Query S3 to get control and treatment json files
+    if not args.platform:
+        raise ValueError("A platform must be specified with the --platform flag to retrieve the "
+                         "previous metrics JSONs as control from S3.")
+    # User only provide the treatement file, and expect us to download from S3
+    control, treatment = None, None
+    if not args.control and args.treatment:
         json_path = Path(args.treatment)
         assert json_path.exists(), f"Specified result json path {args.treatment} does not exist."
-        date: str = get_date_from_metrics(json_path.stem)
+        end_date: datetime = datetime.strptime(get_date_from_metrics(json_path.stem), "%Y-%m-%d")
         userbenchmark_name: str = get_ub_name(args.treatment)
-        latest_metrics_jsons = get_latest_n_jsons_from_s3(1, userbenchmark_name, args.platform, date)
-
-        if len(latest_metrics_jsons) == 0:
-            raise RuntimeWarning("No previous JSONS found to compare against. No regression info has been generated.")
-
-        s3 = S3Client(USERBENCHMARK_S3_BUCKET, USERBENCHMARK_S3_OBJECT)
-        control = s3.get_file_as_json(latest_metrics_jsons[0])
         with open(json_path, "r") as cfptr:
             treatment = json.load(cfptr)
-        regressions_dict = generate_regression_dict(control, treatment)
-        output_path = args.output if args.output else get_default_output_path(control["name"])
-        process_regressions_into_yaml(regressions_dict, output_path, args.control, args.treatment)
-
-        owner = " ".join(args.owner) if args.owner else "@xuzhao9"
-        process_regressions_into_gh_issue(regressions_dict, owner, args.gh_issue_path, args.errors_path)
     else:
-        # S3 path
-        print("Comparison for metrics from Amazon S3 given a userbenchmark name is WIP.")
+        assert args.name, f"To detect regression with S3, you must specify a userbenchmark name."
+        end_date = datetime.strptime(args.end_date, "%Y-%m-%d")
+        userbenchmark_name = args.name
+    available_metrics_jsons = get_latest_n_day_jsons_from_s3(7, userbenchmark_name, args.platform, end_date)
+    # Download control from S3
+    if len(available_metrics_jsons) == 0:
+        raise RuntimeWarning(f"No previous JSONS in a week found to compare towards the end date {end_date}. No regression info has been generated.")
+    print(f"Found metrics json files on S3: {available_metrics_jsons}")
+    start_date = args.start_date if args.start_date else get_best_start_date(available_metrics_jsons, end_date)
```

Comment:
I think the start date should be derived earlier in the code (before retrieving jsons. Then, the ndays could be calculated from start date, and the random 7 from above would not need to occur unless the start date was not specified.

Reply:
We use the user-specified start_date if that's available. Otherwise, we will try to derive it with the retrieved jsons.

Without retrieving jsons, we don't know which date are available (the workflow could fail on some date).